File: chapter4_web_presentation/cgi/cgi-bin/hello.rb
#!/usr/bin/ruby

list = ENV.to_a.map do |key, value|
  "<li>#{key}: #{value}</li>"
end

response = <<EOS
Content-type: text/html

<html>
  <head>
    <title>CGI Example</title>
  </head>
  <body>
    <div>Hello, CGI!</div>
    <ul>
      #{list.join}
    </ul>
  </body>
</html>
EOS

puts(response)

File: chapter4_web_presentation/cgi/cgi-bin/hello.mjs
#!/usr/bin/node

import process from "node:process";

const listItems = Object.keys(process.env).map((key) => {
  const value = process.env[key];
  return `<li>${key}: ${value}</li>`;
});

const response = `
  Content-type: text/html

  <html>
    <head>
      <title>CGI Example</title>
    </head>
    <body>
      <div>Hello, CGI!</div>
      <ul>
        ${listItems.join('')}
      </ul>
    </body>
  </html>
`;

process.stdout.write(response.trim());
